module/zfs/dmu_send.c

```diff
@@ -2824,6 +2861,9 @@ receive_free(struct receive_writer_arg *rwa, struct drr_free *drrf)
 	if (dmu_object_info(rwa->os, drrf->drr_object, NULL) != 0)
 		return (SET_ERROR(EINVAL));
 
+	if (drrf->drr_object > rwa->max_object)
```

Comment:
You don't actually *need* these, since we only consider this value when doing a receive of a full send as a clone, and in that case you have to get an object record before you get any other kind of record for that object. Doesn't hurt to have them, though, and to have it be accurate in all cases.

Reply:
any other kind of record except FREEOBJECT ;) I originally attempted to just re-use last_object, but that obviously did not work out.

I did not feel comfortable enough with my level of understanding of the code to play guessing games (regarding which patterns are possible and which aren't), so I went the "better safe than sorry" route :)